src/olympia/lib/tests/test_git.py

```diff
@@ -860,30 +860,45 @@ def test_get_diff_newline_new_file():
 
     assert len(changes) == 1
     assert changes[0]['new_ending_new_line'] is True
-    assert changes[0]['old_ending_new_line'] is False
+    assert changes[0]['old_ending_new_line'] is True
 
-    hunk_changes = changes[0]['hunks'][0]['changes']
+    assert len(changes[0]['hunks']) == 0
 
-    assert hunk_changes == [
-        {
-            'content': '{"id": "random"}',
-            'new_line_number': -1,
-            'old_line_number': 1,
-            'type': 'delete'
-        },
-        {
-            'content': '\n\\ No newline at end of file',
-            'new_line_number': -1,
-            'old_line_number': 1,
-            'type': 'insert-eofnl'
-        },
-        {
-            'content': '{"id": "random"}',
-            'new_line_number': 1,
-            'old_line_number': -1,
-            'type': 'insert'
-        }
-    ]
+
+@pytest.mark.django_db
+def test_get_diff_eol_changes():
+    addon = addon_factory(file_kw={'filename': 'webextension_no_id.xpi'})
+
+    original_version = addon.current_version
+
+    AddonGitRepository.extract_and_commit_from_version(original_version)
+
+    parent_version = version_factory(
+        addon=addon, file_kw={'filename': 'webextension_no_id.xpi'})
+
+    repo = AddonGitRepository.extract_and_commit_from_version(parent_version)
+
+    # Let's replace the unix eol with dos ones.
+    apply_changes(
+        repo, parent_version,
+        '{"id": "random",\r\n"name": "foo"\r\n}', 'manifest.json')
+
+    version = version_factory(
+        addon=addon, file_kw={'filename': 'webextension_no_id.xpi'})
+
+    repo = AddonGitRepository.extract_and_commit_from_version(version)
+
+    # Now go back to unix eol.
+    apply_changes(
+        repo, version,
+        '{"id": "random",\n"name": "foo"\n}', 'manifest.json')
+
+    changes = repo.get_diff(
+        commit=version.git_hash,
+        parent=parent_version.git_hash)
+
+    assert len(changes) == 1
+    assert len(changes[0]['hunks']) == 0
```

Comment:
Does that make sense, @EnTeQuAk ? Not sure why I still get a change with no hunks.

Reply:
I think that's fine, do any other properties in the `changes` change? I wonder if this behaves differently if you're using `GIT_DIFF_IGNORE_WHITESPACE` rather than just `GIT_DIFF_IGNORE_WHITESPACE_CHANGE` (which then ignores whitespace changes at the end of line too)

Reply:
```
> pprint(changes)                                                                                                                                                                
[{'hash': 'f28ce3675d7aa63548560f2883bbde2aac8c95f3',
  'hunks': [],
  'is_binary': False,
  'lines_added': 0,
  'lines_deleted': 0,
  'mode': 'M',
  'new_ending_new_line': True,
  'old_ending_new_line': True,
  'old_path': 'manifest.json',
  'parent': 'e6636d850a0b92367924745b029fad93725b4d98',
  'path': 'manifest.json',
  'size': 17}]
```

Reply:
It makes sense I think: the file *has* been modified, we just happen to ignore all those modifications when showing the diff, but it's still modified as far as git is concerned.

Reply:
> It makes sense I think: the file has been modified, we just happen to ignore all those modifications when showing the diff, but it's still modified as far as git is concerned.

I think that explains it pretty good. Mind putting that into a comment in the test for us in a few months time?

Reply:
It did, `size` reflects the actual size of the file contents.